Code/GraphMol/SynthonSpaceSearch/MemoryMappedFileReader.h

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (C) David Cosgrove 2025.
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#ifndef MEMORYMAPPEDFILEREADER_H
+#define MEMORYMAPPEDFILEREADER_H
+
+#include <string>
+
+#include <RDGeneral/export.h>
+
+namespace RDKit::SynthonSpaceSearch::details {
+
+struct RDKIT_SYNTHONSPACESEARCH_EXPORT MemoryMappedFileReader {
+  MemoryMappedFileReader() = delete;
+  MemoryMappedFileReader(const std::string &filePath);
+  MemoryMappedFileReader(const MemoryMappedFileReader &) = delete;
+  MemoryMappedFileReader(MemoryMappedFileReader &&) = delete;
```

Comment:
It's ok to delete this, but I think that move operations should work fine with this class: you just want to make sure you zero out the pointer in the instance being moved from.